### ISSUE TITLE
Training consistently produces worse error and is rolled back — 217 'Training X caused a higher error' events per run

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#2382]
## Summary

`GRQ-22-sloth.log` logs `Training <id> caused a higher error of <new> from <old>` followed by `… and no tuning` 217 times across a single ~40-minute evolve run. Every training cycle for the symbols in this run produces a net regression that is then silently rolled back. Training is supposed to *improve* the creature; a 100% regression rate indicates the training step is either miscalibrated, acting on the wrong weight set, or colliding with a concurrent mutation.

## What Needs to Be Done

1. Reproduce the regression on the reference creature (`2054 observations, 110 neurons, 3 outputs, 3141–3216 synapses`) used in the log and capture before/after observations for each training step.
2. Identify the root cause. Likely suspects:
   - Learning rate / elastic distribution step size is too large given the current weight magnitudes (see also #2378 — huge weights produce huge gradients that overshoot).
   - Training is operating on a creature state that already had a weight mutation applied but the fitness baseline is from the pre-mutation state, making every training step look like a regression.
   - The `target error: 0.05, iterations: 1` budget is too tight to observe any improvement — single-iteration training with a learning-rate-decay schedule designed for many iterations will usually worsen fitness.
3. Either:
   - Adapt the training schedule (reduce learning rate / allow more iterations) so the common case is an improvement rather than a regression, or
   - Detect the "training always regresses" pattern early and skip training for that creature/generation instead of spending cycles producing a rollback.
4. Add a structured metric (training-regression rate per generation) to the existing training lifecycle logging (#1615 was closed — verify the event stream still exposes enough detail).

## Acceptance Criteria

- [ ] Regression reproducible in a unit or benchmark test against a fixture modelled on the log's reference creature.
- [ ] Root cause documented in the PR summary with before/after numbers.
- [ ] After the fix, the training-regression rate on the reference workload is below 25% per run (current is ~100%), or training is skipped for creatures where regression is predicted.
- [ ] New tests covering the happy path (training improves fitness), the rollback path (training worsens fitness, no state change), and the skip path (predicted regression, training bypassed).
- [ ] `./quality.sh` passes.

## Dependencies

Depends on #2378 — weight/bias magnitudes reaching `1e+195` will produce meaningless gradients; fixing that first may eliminate a large portion of the regressions.

## Context

Part of #2379

Representative evidence from `https://github.com/stSoftwareAU/GRQ-logs/blob/Develop/GRQ-22-sloth.log`:

```
Training fd59fea8 caused a higher error of 0.061729328704958175 from 0.06172921886819426
Training fd59fea8 caused a higher error of 0.061729328704958175 from 0.06172921886819426 and no tuning
…
Training 66260aa7 caused a higher error of 0.14146075969779812 from 0.14112731134301545
Training 66260aa7 caused a higher error of 0.14146075969779812 from 0.14112731134301545 and no tuning
```

`grep -c 'caused a higher error'` on the log returns 217.

**Comment by stservice** [comment]
Claimed by `stservice-1776797482736`

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/2388 has been created to address this issue.
